### PR TITLE
Another attempt at AEI

### PR DESCRIPTION
**Pull request by henrymoss**
This another (hopefully final) attempt at Augmented Expected Improvement (AEI).

This time I have tweaked the model wrappers to collect observation noise variance (when availible), rather than just trying to access raw model.model....

Access to observation noise is required for my next 2 PRs (GIBBON and RFF-based Thompson sampling).

### CODE REVIEW

File: trieste/models/model_interfaces.py [hunk 1]
```diff
@@ -379,6 +399,19 @@ def covariance_between_points(
...
 
+    def get_observation_noise(self):
+        """
+        Return the variance of observation noise for homoscedastic likelihoods.
+        :return: The observation noise.
+        :raise NotImplementedError: If the model does not have a homoscedastic likelihood.
+        """
+        try:
+            noise_variance = self.model.likelihood.variance
+        except NotImplementedError:
```
Comment (uri-granta) [inline trieste/models/model_interfaces.py]: Are you sure this is right? Do GPModels always have a likelihood.variance? If not, does trying to access it result in a NotImplementedError or an AttributeError?

File: trieste/acquisition/function.py [hunk 2]
```diff
@@ -156,6 +156,85 @@ def acquisition(x: TensorType) -> TensorType:
...
+    improvement moves towards the objective function's minimum, and the expectation is calculated
+    with respect to the ``model`` posterior. In contrast to standard EI, AEI has an additional
+    multiplicative factor that penalizes evaluations made in areas of the space with very small
+    posterior predictive variance. Thus, when applying standard EI to noisy optimisation
+    problems, AEI avoids getting trapped and repeatedly querying the same point.
+    For model posterior :math:`f`, this is
+    .. math:: x \mapsto EI(x) * \left(1 - frac{\tau^2}{\sqrt{s^2(x)+\tau^2}}\right),
+    where :math:`s^2(x)` is the predictive variance and :math:`\tau` is observation noise.
+    This function was introduced by Huang et al, 2006. See :cite:`Huang:2006` for details.
+    :param model: The model of the objective function.
```
Comment (uri-granta) [inline trieste/acquisition/function.py:204]: Missing blank line before :param

**Review by uri-granta — APPROVED** [review]
LGTM

File: trieste/models/model_interfaces.py (hunk 1 above)
Comment (henrymoss) [reply trieste/models/model_interfaces.py]: You are right, have made changes

**Comment by henrymoss** [comment]
I cant get the docs to pass :( Is there a way to get more intuitive error messages @uri-granta ?

**Pull request merged**